rclpy/rclpy/node.py

```diff
@@ -31,6 +31,19 @@
 from rclpy.validate_topic_name import validate_topic_name
 
 
+def check_for_type_support(msg_type):
+    try:
+        if msg_type.__class__._TYPE_SUPPORT is None:
+            msg_type.__class__.__import_type_support__()
+    except AttributeError:
+        raise AttributeError(
```

Comment:
By raising a new exception the stacktrace of the previous one is being lost. Instead you could update the raised exception instead.

Reply:
Yeah is did it because `AttributeError` doesnt have a `msg` attribute that I could extend. I could try to just overwrite its args but looks like args have to be 1char long so it will give a pretty ugly result.

How would you recommend to do it ?

Reply:
`e.args` is a `tuple` with one element which is the message string. You can simply reassign `args` to point to your custom `tuple`. Since `args` might contains more than one element being safe is better than being sorry in the future:

```
e.args = (e.args[0] + ' foo', *e.args[1:])
```

Reply:
oh yeah totally make sense. Thanks for the explanation! Fixed in https://github.com/ros2/rclpy/pull/117/commits/72db1490b4de31b963d431728fddb5fee4f39fc5